connector/src/main/scala/com/google/cloud/spark/bigquery/direct/DirectBigQueryRelation.scala

```diff
@@ -269,10 +267,19 @@ private[bigquery] class DirectBigQueryRelation(
   def getMaxNumPartitionsRequested: Int =
     getMaxNumPartitionsRequested(defaultTableDefinition)
 
-  def getMaxNumPartitionsRequested(tableDefinition: StandardTableDefinition): Int =
+  def getMaxNumPartitionsRequested(tableDefinition: TableDefinition): Int =
     options.maxParallelism
       .getOrElse(Math.max(
-        (tableDefinition.getNumBytes / DEFAULT_BYTES_PER_PARTITION).toInt, 1))
+        (getNumBytes(tableDefinition) / DEFAULT_BYTES_PER_PARTITION).toInt, 1))
+
+  def getNumBytes(tableDefinition: TableDefinition): Long = {
+    val tableType = tableDefinition.getType
+    if (options.viewsEnabled && TableDefinition.Type.VIEW == tableType) {
+      0
```

Comment:
Please replace the value with the default: `sqlContext.conf.defaultSizeInBytes` (taken from [here](https://github.com/apache/spark/blob/v2.4.0/sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala#L212))
Based on the documentation having a size of 0 may have side effects.

Reply:
`sqlContext.conf` is not accessible outside of `org.apache.spark.sql` so I had to use `sqlContext.sparkSession.sessionState.conf.defaultSizeInBytes` instead